### DISCUSSION TITLE
Watermarking with CUPS-Filter

### DISCUSSION THREAD

**Discussion by FritzscheAnita** [OpenPrinting/cups-filters#462] [Q&A]
Since watermarking prints is not supported i tried to implement a cupsFilter for doing it manually. 
I have two ppd-files i tried to integrate the filter in. One still supports the cupsFilter as it is. The watermarking basically works. However the pstopdf does not the original document is ignored and just the watermark is printed. (But nevermind that)
In the second, ppd which just came with the driverless printing in /etc/cups/ppd cupsFilter2 is used, so the original cupsFilter line does not work. 
My question is How do i write the same line correctly for cupsFilter2 and where exactly do I place it?
```
*cupsFilter: "application/vnd.cups-postscript 100 watermark"
*cupsFilter2: "application/vnd.cups-postscript application/postscript 100 watermark"
```

Is there any smarter way to integrate watermarks?

**Comment by pdfkungfoo** [comment]
I'd be interested to see your actual *`watermark`* filter code. Is it a script employing something like `pdftk` or `qpdf`?

The `*cupsFilter:` line can go anywhere in the first 5-40 lines of the PPD file I'd say. Just not in the middle of another logical "block", like `*OpenUI... *CloseUI`

Also, I'd change the line to this (but that's not very highly relevant):
~~~~
*cupsFilter2: "application/vnd.cups-postscript application/vnd.cups-postscript 100 watermark"
~~~~
